svelte-kit/src/lib/components/ButtonIcon.svelte

```diff
@@ -0,0 +1,42 @@
+<script lang="ts">
+    import Fa from 'svelte-fa/src/fa.svelte';
+    import {
+        faCheck,
+        faPenToSquare,
+        faPlus,
+        faTrashCan,
+        faXmark
+    } from '@fortawesome/free-solid-svg-icons';
+
+    export let action: () => void;
+    export let classes = '';
+    export let padding = 4;
+    export let variant = 'add';
+
+    let color: string;
+    let icon: any;
+
+    if (variant === 'add') {
+        color = 'primary';
+        icon = faPlus;
+    } else if (variant === 'edit') {
+        color = 'primary';
+        icon = faPenToSquare;
+    } else if (variant === 'delete') {
+        color = 'warning';
+        icon = faTrashCan;
+    } else if (variant === 'ok') {
+        color = 'tertiary';
+        icon = faCheck;
+    } else if (variant === 'cancel') {
+        color = 'warning';
+        icon = faXmark;
+    }
```

Comment:
A way more typescript-friendly version :

```suggestion
    type Variant = {
        color: string,
        icon: string
    }

    type Variants = {
        add: Variant,
        edit: Variant,
        delete: Variant,
        ok: Variant,
        cancel: Variant,
    };

    export let variant: keyof Variants;

    const variants: Variants = {
        add: {
            color: 'primary',
            icon : 'faPlus'
        },
        // ...
    };

    $: color = variants[variant].color ?? 'primary';
    $: icon = variants[variant].icon ?? 'faPlus';
```

This will allow dynamic allocation of color and icon while providing good type support for a better dev experience.